src/crimson/osd/pg.cc

```diff
@@ -661,7 +661,24 @@ seastar::future<Ref<MOSDOpReply>> PG::handle_failed_op(
   }, load_obc_ertr::assert_all{ "can't live with object state messed up" });
 }
 
-seastar::future<Ref<MOSDOpReply>> PG::do_osd_ops(
+crimson::errorator<crimson::ct_error::eagain>::future<> PG::rep_repair_primary_object(
+  Ref<MOSDOp> m,
+  const hobject_t& oid,
+  eversion_t& v) 
+{
+  ceph_assert(is_primary());
```

Comment:
please use `ceph_assert()` only for verifying the runtime status or return value. in this case, i'd suggest use `assert()` instead, for documenting and verifying the programming preconditoins.